### ISSUE TITLE
Update REST API for settings secrets

### ISSUE THREAD

**Issue by dfaller** [ngageoint/scale#816]
Update the REST API to handle job configuration (settings/mounts) with secret values.

1. Create a validate() method for JobConfiguration that takes in a JobInterface, strips out any settings or mounts that are not in the interface, removes any settings that are secret, and returns a warning describing each setting/mount removed and why. See similar validate() methods for examples.
2. Add configuration to job type detail calls - GET /job-types/{id}/
Make sure to call JobConfiguration.validate() during job type detail calls. This will ensure that returned settings/mounts match any interface changes. Update REST API docs and add unit tests to ensure invalid/secret settings/mounts are not returned.
3. Add configuration to job type validation - POST /job-types/validation/
Make sure to call JobConfiguration.validate() and return warnings. Update REST API docs and add unit tests testing the validation.
4. Add configuration to job type edit - PATCH /job-types/{id}/
If there are secret settings passed in the configuration field, set these values in Vault. Make sure to call JobConfiguration.validate() before saving the JSON into the configuration field. Do calls to Vault after updating job type model, but within database transaction so Vault failure will roll back database changes. Update REST API docs and add unit tests (mock out Vault calls). Also please add the shared_memory_required field (this might just be a doc update).
5. Add configuration to job type import - POST /configuration/
Basically the same as (4), make them both go through the same code path (one way to edit a job type).
6. Add configuration to job type import validation - POST /configuration/validation/
Basically the same as (3), again, one code path
7. Add configuration to job type export - GET /configuration/
Basically the same as (2), one code path
